### DISCUSSION TITLE
huge mp3 files and ffmpeg

### DISCUSSION THREAD

**Discussion by trombik** [music-assistant/support#938] [Q&A] [answered]
hi,

installed `mass` yesterday, and it almost works. my question is: what would happen when local files are huge mp3 files? they are like between 200MB and 500MB.

I have been seeing  multiple `ffmpeg` processes are stacked in the process table when i click "play next". the next file is played, but the `ffmpeg` process that is probably processing the previous one is still running. eventually, the `hass` is killed due to out of swap. the machine is not so powerful, it has 2GB of RAM. the CPU is not powerful either, Atom 230 @1.6Ghz. 

IIRC, the plain `hass` music player just passes a URL to the device (in my case, Google Home). is it possible to do the same with `mass`?

**Comment by marcelveldt** [comment, marked as answer]
files are read in chunks so no worries about large files. I test this using very large hi res files too so you're talking about gigabytes of data.

Multiple ffmpeg processes is fine. It takes a few seconds to stop the previous stream when the next is already started and each stream session can exist of multiple ffmpeg instances due to various stages in the processing.

It works fine on a pi 4 with only 1 GB of memory so it should also work in your more powerful setup.
If not, something is wrong and you should perhaps file an issue report. I assume you are running supported HA environment like the official HA Container or HAOS ?

And lastly, MA sends an URL to the device too, that of its own stream engine.

**Reply by trombik** [reply]
@marcelveldt thanks.

> It works fine on a pi 4 with only 1 GB of memory so it should also work in your more powerful setup. If not, something is wrong and you should perhaps file an issue report. I assume you are running supported HA environment like the official HA Container or HAOS ?

that's good to know. no, my environment is NOT "supported HA environment" (`virtualenv` on FreeBSD). that's why i asked here.

it might be caused by other processes on the same machine, then.

**Reply by marcelveldt** [reply]
FreeBSD is not supported by both HA and MA. I'd advice you to run it on other hardware or some VM or something.

**Reply by trombik** [reply]
I know. but it is not allowed in my grandmother's will.

**Reply by marcelveldt** [reply]
I'm very sorry but I can't help you then. We're already busy enough supporting the officially supported setups.

**Comment by trombik** [comment]
your explanation already helped a lot to understand how it works. thanks.

**Comment by OzGav** [comment]
Relates to v1
